tools/src/main/scala/caliban/tools/SchemaWriter.scala

```diff
@@ -183,6 +209,42 @@ object SchemaWriter {
       }
     }
 
+    def writeNewTypeClasses(fieldType: Type, directive: Directive) = {
+      @tailrec def writeTypeOf(fieldType: Type): String = fieldType match {
+        case NamedType(name, _) => scalarMappings.flatMap(_.get(name)).getOrElse(name)
+        case ListType(ftype, _) => writeTypeOf(ftype)
+      }
+
+      val fnName  = safeName(directive.arguments("name").toInputString.replace("\"", ""))
+      val newtype = safeName(writeTypeOf(fieldType))
+
+      s"""case class $fnName(value : $newtype) extends AnyVal
+         |object $fnName {
+         |    implicit val schema: Schema[Any, $fnName] = summon[Schema[Any, $newtype]].contramap(_.value)
```

Comment:
I'm a bit confused, does `summon` work for Scala 2?

Also, perhaps just use `Schema[$newtype].contramap(_.value)` instead, the `Schema.apply` method "summons" the implicit `Schema`

Reply:
Could not get that to work, but used `implicitly` (Scala 2 version of summon).
Tested it on my Scala 3 setup and that worked, but have to double check my Scala options.

Reply:
To catch these kind of issues it would be nice to add the new annotation to the sbt plug-in "scripted" test suite because it runs both on scala 2 and scala 3